http2_general/test_h2_block_action.py

```diff
@@ -78,13 +58,14 @@ def check_last_error_response(self, client, expected_status_code, expected_goawa
         response body we also can't send response, because we need to process
         WINDOW_UPDATE frames, but can't do it on the DEAD sock.
         """
-        if not run_config.TCP_SEGMENTATION and (
-            self.INITIAL_WINDOW_SIZE > len(self.ERROR_RESPONSE_BODY)
-        ):
-            self.assertTrue(client.wait_for_response())
-            self.assertEqual(client.last_response.status, expected_status_code)
-            self.assertEqual(client.last_response.body, self.ERROR_RESPONSE_BODY)
-            self.assertIn(expected_goaway_code, client.error_codes)
+        if not run_config.TCP_SEGMENTATION:
```

Comment:
Perhaps we shouldn't run this test with TCP segmentation? So you can remove this `if`.